airflow/providers/amazon/aws/utils/connection_wrapper.py

```diff
@@ -154,6 +154,16 @@ def __post_init__(self, conn: "Connection"):
 
         self.endpoint_url = extra.get("host")
 
+        if self.endpoint_url:
+            warnings.warn(
+                "extra['host'] is deprecated and will be removed in a future release."
```

Comment:
@potuik If the [Amazon provider is going to be a major version bump](https://lists.apache.org/thread/c9254qscbf6yjpqx6q4n3o798pxxyfwk) this month, is the deprecation message even required? The provider could just stop supporting the `host` connection field.

Reply:
Yeah. That's a bit aggressive, but we could  raise plain error here. I do not see this as being a major problem. But removing it altogether is not a good idea because it would likely stop doing that it was doing before without even a warning in some cases.

Reply:
Understood. So more of a release flow of `feature` -> `deprecate/message about not using feature` -> `remove feature` regardless of major version bump along the way.